src/main/scala/rocketchip/Generator.scala

```diff
@@ -86,10 +91,80 @@ trait Generator extends App with HasGeneratorUtilities {
     writeOutputFile(td, s"${names.configs}.knb", world.getKnobs) // Knobs for DSE
     writeOutputFile(td, s"${names.configs}.cst", world.getConstraints) // Constraints for DSE
     ConfigStringOutput.contents.foreach(c => writeOutputFile(td, s"${names.configs}.cfg", c)) // String for software
+
+    params(SynTopName) match {
+      case Some(synTopName) => {
+        firrtl.Driver.compile(
+          s"${names.targetDir}/$longName.fir",
+          s"${names.targetDir}/$longName.TestHarness.v",
+          new EmitHarnessVerilog(synTopName),
+          Parser.UseInfo
+        )
+
+        firrtl.Driver.compile(
+          s"${names.targetDir}/$longName.fir",
+          s"${names.targetDir}/$longName.v",
+          new EmitTopVerilog(synTopName),
+          Parser.UseInfo,
+          AnnotationMap(Seq(
```

Comment:
I assume all these firrtl invocations came from the firrtl repo and are therefore correct.